staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -916,6 +917,26 @@ func (r *Request) newHTTPRequest(ctx context.Context) (*http.Request, error) {
 	return req, nil
 }
 
+// newDNSMetricsTrace returns an HTTP trace that tracks time spend on DNS lookups per network/host.
+// This metric is available in client as "rest_client_resolver_duration_seconds".
+func newDNSMetricsTrace(ctx context.Context) *httptrace.ClientTrace {
+	var (
+		dnsStart time.Time
+		dnsHost  string
+	)
+	return &httptrace.ClientTrace{
+		DNSStart: func(info httptrace.DNSStartInfo) {
+			dnsStart = time.Now()
+			dnsHost = info.Host
+		},
+		DNSDone: func(info httptrace.DNSDoneInfo) {
+			for i := range info.Addrs {
+				metrics.ResolverLatency.Observe(ctx, info.Addrs[i].Network(), dnsHost, time.Since(dnsStart))
+			}
```

Comment:
```suggestion
				metrics.ResolverLatency.Observe(ctx, dnsHost, time.Since(dnsStart))
```

you only need the latency of the host, not of the resolved addresses